src/main/scala/org/broadinstitute/dsde/workbench/avram/dataaccess/HttpSamDao.scala

```diff
@@ -0,0 +1,32 @@
+package org.broadinstitute.dsde.workbench.avram.dataaccess
+
+import java.util.logging.Logger
+
+import com.softwaremill.sttp._
+import io.circe.Decoder.Result
+import io.circe.{DecodingFailure, Json, ParsingFailure}
+import io.circe.generic.auto._
+import io.circe.parser._
+import org.broadinstitute.dsde.workbench.avram.util.{ErrorResponse, RestClient}
+
+class HttpSamDao(samUrl: String) extends SamDao with RestClient {
+  private val log: Logger = Logger.getLogger(getClass.getName)
+
+  override def getUserStatus(token: String): Either[ErrorResponse, SamUserInfoResponse] = {
+    val request = buildAuthenticatedGetRequest(samUrl, "/register/user/v2/self/info", token)
+    val response: Id[Response[String]] = request.send()
+    response.body match {
+      case Left(error: String) => Left(ErrorResponse(response.code, error))
```

Comment:
I'm wondering if Avram should define its own exceptions+messages instead of bubbling up underlying error messages to the user. The Sam one in particular might be pretty verbose containing a JSON stack trace which the user doesn't need to see.

Leo does this by defining a `LeoException` trait and mapping underlying errors to a `LeoException` implementation with a user friendly status code and message. 

If you map errors to `AvramException` it might help with monadic use of `Either`, too.

Reply:
I'll have a look at `LeoException`. Good point about immediately translating external errors into a single error catcher type potentially helping with `Either`.